test/python/test_qasm_parser.py

```diff
@@ -78,5 +81,37 @@ def test_get_tokens(self):
             self.assertTrue(isinstance(token, ply.lex.LexToken))
 
 
+class TestParserWithComments(QiskitTestCase):
+    """QasmParser has a with_comment option to get have comments as extra nodes"""
+
+    def test_single_comment(self):
+        """The thing to parse is a single comment"""
+        qasm_string = """//this is a comment"""
+        ast = Qasm(data=qasm_string).parse(with_comments=True)
+        self.assertEqual(len(ast.children), 1)
+        self.assertIsInstance(ast.children[0], Comment)
+        self.assertEqual(ast.qasm().replace('\n', '', 1), qasm_string)
+
+    def test_inline_comment(self):
+        """The thing to parse is a single comment"""
```

Comment:
I think this docstring is a repetition of the previous one. Can we update it?

Reply:
fixed in 7068fbd